scheduler/src/cook/kubernetes/api.clj

```diff
@@ -216,25 +257,42 @@
   [task-uuid]
   "TODO: Implement task-uuid->pod-name") ; TODO
 
-(defn remove-finalization-if-set-and-delete
-  "Remove finalization for a pod if its there. No-op if its not there."
-  [api-client ^V1Pod pod]
-  (TODO))
-
 (defn kill-task
   "Kill this kubernetes pod"
   [^ApiClient api-client ^V1Pod pod]
-  (let [api (CoreV1Api. api-client)]
-    (.deleteNamespacedPod ; TODO: Double check arguments.
-      api
-      (-> pod .getMetadata .getName)
-      (-> pod .getMetadata .getNamespace)
-      nil
-      nil
-      nil
-      nil
-      nil
-      nil)))
+  (let [api (CoreV1Api. api-client)
+        ^V1DeleteOptions deleteOptions (-> (V1DeleteOptionsBuilder.) (.withPropagationPolicy "Background") .build)
+        ]
+    (try
+      (.deleteNamespacedPod ; TODO: Double check arguments.
+        api
+        (-> pod .getMetadata .getName)
+        (-> pod .getMetadata .getNamespace)
+        deleteOptions
+        nil
+        nil
+        nil
+        nil
+        nil)
+      (catch JsonSyntaxException e
```

Comment:
Can we log this or have a metric for it? It'd be nice to have some way of noticing a change in rate.

Reply:
Logging it now.